openreview/conference/builder.py

```diff
@@ -138,23 +138,30 @@ def close_submissions(self, freeze_submissions = True):
 
         # Get invitation
         invitation = self.client.get_invitation(id = self.get_submission_id())
-        if invitation.reply['writers'] != self.id:
-            invitation.reply['writers'] = self.id
 
+        # Set duedate in the past
         now = round(time.time() * 1000)
         if invitation.duedate > now:
             invitation.duedate = now
-        self.client.post_invitation(invitation)
+        invitation = self.client.post_invitation(invitation)
 
-        # if freeze submissions then remove writers
+        # If freeze submissions then remove writers
         # use a process pool to run this
         if freeze_submissions:
+            if invitation.reply['writers'] != self.id:
+                invitation.reply['writers'] = self.id
+                invitation = self.client.post_invitation(invitation)
+
             notes_iterator = tools.iterget_notes(self.client, invitation = invitation.id)
             for note in notes_iterator:
                 if note.writers != [self.id]:
                     note.writers = [self.id]
                     self.client.post_note(note)
 
+        # Add venue to active venues
+        active_venues_group = self.client.get_group(id = 'active_venues')
+        self.client.add_members_to_group(active_venues_group, [self.get_id()])
```

Comment:
I wonder if this should be done more explicitly..? what do you think?

Reply:
Don't we want to add the venue to active venues once the submission is closed, or do you think it should be optional?

Reply:
maybe it should be optional. on the other hand, it's easy to change, and we'll find out if it's necessary to make it optional later.